Networking/NetworkingTests/Remote/LeaderboardsRemoteTests.swift

```diff
@@ -0,0 +1,75 @@
+import XCTest
+@testable import Networking
+
+/// Leaderboards Unit Tests
+///
+final class leaderboardsRemoteV4Tests: XCTestCase {
+
+    let network = MockupNetwork()
+    let sampleSiteID: Int64 = 1234
+
+    override func setUp() {
+        super.setUp()
+        network.removeAllSimulatedResponses()
+    }
+
+    func testLeaderboardReturnsCorrectParsedValues() throws {
+        // Given
+        let remote = LeaderboardsRemote(network: network)
+        network.simulateResponse(requestUrlSuffix: "leaderboards", filename: "leaderboards-year")
+
+        // When
+        var remoteResult: Result<[Leaderboard], Error>?
+        waitForExpectation { exp in
+            remote.loadLeaderboards(for: sampleSiteID,
+                                    unit: .yearly,
+                                    earliestDateToInclude: "2020-01-01T00:00:00",
+                                    latestDateToInclude: "2020-12-31T23:59:59",
+                                    quantity: 3) { result in
+                                        remoteResult = result
+                                        exp.fulfill()
+            }
+        }
+
+        // Then
+        let leaderboards = try XCTUnwrap(remoteResult?.get())
+
+        // API Returns 4 leaderboards
+        XCTAssertEqual(leaderboards.count, 4)
+
+        // The 4th leaderboard contains the top products and should not be empty
+        let topProducts = leaderboards[3]
+        XCTAssertFalse(topProducts.rows.isEmpty)
+
+        // Each prodcut should have non-empty values
+        topProducts.rows.forEach { product in
+            XCTAssertFalse(product.subject.display.isEmpty)
+            XCTAssertFalse(product.subject.value.isEmpty)
+            XCTAssertFalse(product.quantity.display.isEmpty)
+            XCTAssertTrue(product.quantity.value > 0)
+            XCTAssertFalse(product.total.display.isEmpty)
+            XCTAssertTrue(product.total.value > 0)
```

Comment:
My rationale was that this may be easier to maintain in cases where we updated the mock JSON file.
But I can add the concrete assert if you prefer them!

Reply:
I think the actual value is preferred since we don't expect the mock JSON file to change a lot. And it will help catch numerical JSON conversion. Unlikely to fail but still. 🤷

Reply:
Also for future reference → [XCTAssertGreaterThanOrEqual](https://developer.apple.com/documentation/xctest/xctassertgreaterthanorequal) exists. 🙂

Reply:
Done 556c4b43a6fd530d5aeb4b4e9cacc53ac8c40849